spec/features/responses_index_spec.rb

```diff
@@ -25,22 +29,66 @@
     end
   end
 
-  context "with key question" do
-    let(:user) { create(:user) }
-    let(:form) { create(:form, :published, question_types: %w[text]) }
-    let!(:response1) { create(:response, user: user, form: form, answer_values: ["pants"]) }
-    let!(:response2) { create(:response, user: user, form: form, answer_values: ["sweater"]) }
+  describe "existing responses" do
+    let!(:response) do
+      create(
+        :response,
+        user: user,
+        form: form,
+        reviewed: true,
+        answer_values: ["pants in i-am-a-banana"]
+      )
+    end
+
+    context "with key question" do
+      before { form.c[0].question.update!(key: true) }
 
-    before do
-      form.c[0].question.update!(key: true)
+      scenario "key question values are shown in index" do
+        click_link("Responses")
+        expect(page).to have_content(form.c[0].code)
+        expect(page).to have_content("pants")
+        expect(page).to have_content("Responses")
+      end
     end
 
-    scenario "key question values are shown in index" do
-      login(user)
-      click_link("Responses")
-      expect(page).to have_content(form.c[0].code)
-      expect(page).to have_content("pants")
-      expect(page).to have_content("sweater")
+    context "search" do
+      describe "with answer text" do
+        scenario "works" do
+          fill_in "search_str", with: "pants"
+          click_on "Search"
+
+          # a scoped responses index page shows
+          expect(page).to have_content("Responses")
+          expect(page).to have_content(response.shortcode.upcase)
+          expect(current_url).to end_with("/responses?search=pants")
+        end
+      end
+
+      describe "with short code" do
+        before { response.update(shortcode: "i-am-a-banana") }
+
+        scenario "for user that can edit response" do
+          fill_in "search_str", with: response.shortcode
+          click_on "Search"
+
+          # the response edit page shows
+          expect(page).to have_content("Edit Response")
+          expect(current_url).to end_with("responses/#{response.shortcode}/edit")
+        end
+
+        describe "enumerator" do
+          let(:user) { create(:user, role_name: :enumerator) }
+
+          scenario "for user that can not edit response" do
+            fill_in "search_str", with: response.shortcode
+            click_on "Search"
+
+            # the response show page shows
+            expect(page).to have_content("Response: #{response.shortcode.upcase}")
+            expect(current_url).to end_with("responses/#{response.shortcode}")
+          end
+        end
+      end
```

Comment:
Specs look great! Let's pair real quick so I can show you a few bits of polish (please book me) but you've got the idea!